templates/distribution/manifests/dr/kustomization.yaml.tpl

```diff
@@ -41,13 +61,65 @@ patchesStrategicMerge:
   - patches/velero-schedule-manifests.yml
   - patches/velero-schedule-full.yml
 {{- end }}
+{{- if eq .spec.distribution.common.provider.type "none" }}
+{{- if ne .spec.distribution.modules.dr.etcdBackup.type "none" }}
+  - patches/etcd-backup-schedule.yml
+{{- end }}
+{{- end }}
+
+{{- if eq .spec.distribution.common.provider.type "none" }}
+{{- if ne .spec.distribution.modules.dr.etcdBackup.type "none" }}
+configMapGenerator:
+{{- if eq .spec.distribution.modules.dr.etcdBackup.type "all" "s3" }}
+  - name: etcd-backup-s3-config
+    behavior: replace
+    literals:
+      - target={{ print "minio:" .spec.distribution.modules.dr.etcdBackup.s3.bucketName }}
+      - retention={{ .spec.distribution.modules.dr.etcdBackup.s3.retentionTime }}
+    {{- if eq (index .spec.distribution.modules.dr.etcdBackup "backupPrefix") nil }}
+      - backup-prefix={{ print .metadata.name "-" }}
+    {{- else}}
+      - backup-prefix={{ .spec.distribution.modules.dr.etcdBackup.backupPrefix }}
+    {{- end}}
+  - name: etcd-backup-s3-certificates-location
+    behavior: replace
+    literals:
+      - ETCDCTL_CACERT=/etcd/etcd/ca.crt
+      - ETCDCTL_CERT=/etcd/apiserver-etcd-client.crt
+      - ETCDCTL_KEY=/etcd/apiserver-etcd-client.key
+{{- end }}
+{{- if eq .spec.distribution.modules.dr.etcdBackup.type "all" "pvc" }}
+  - name: etcd-backup-pvc-config
+    behavior: replace
+    literals:
+      - retention={{ .spec.distribution.modules.dr.etcdBackup.persistentVolumeClaim.retentionTime }}
+    {{- if eq (index .spec.distribution.modules.dr.etcdBackup "backupPrefix") nil }}
+      - backup-prefix={{ print .metadata.name "-" }}
+    {{- else}}
+      - backup-prefix={{ .spec.distribution.modules.dr.etcdBackup.backupPrefix }}
+    {{- end}}
+  - name: etcd-backup-pvc-certificates-location
+    behavior: replace
+    literals:
+      - ETCDCTL_CACERT=/etcd/etcd/ca.crt
+      - ETCDCTL_CERT=/etcd/apiserver-etcd-client.crt
+      - ETCDCTL_KEY=/etcd/apiserver-etcd-client.key
+{{- end }}
+{{- end }}
+{{- end }}
 
 {{- if eq .spec.distribution.common.provider.type "none" }}
-{{- if eq .spec.distribution.modules.dr.velero.backend "externalEndpoint" }}
 secretGenerator:
+{{- if eq .spec.distribution.modules.dr.velero.backend "externalEndpoint" }}
```

Comment:
notice that with this change, you can end up with a `secretGenerator:` line that has nothing below. I think kustomize handles this gracefully, but double check it if you haven't.